server/sonar-server/src/main/java/org/sonar/server/issue/IssueQueryService.java

```diff
@@ -174,127 +183,27 @@ public IssueQuery createFromRequest(Request request) {
-      if (Qualifiers.VIEW.equals(uniqueQualifier) || Qualifiers.SUBVIEW.equals(uniqueQualifier)) {
-        List<String> filteredViewUuids = newArrayList();
-        for (String viewUuid : allComponentUuids) {
-          if ((Qualifiers.VIEW.equals(uniqueQualifier) && UserSession.get().hasProjectPermissionByUuid(UserRole.USER, viewUuid))
-            || (Qualifiers.SUBVIEW.equals(uniqueQualifier) && UserSession.get().hasComponentUuidPermission(UserRole.USER, viewUuid))) {
-            filteredViewUuids.add(viewUuid);
-          }
-        }
-        if (filteredViewUuids.isEmpty()) {
-          filteredViewUuids.add(UNKNOWN);
-        }
-        builder.viewUuids(filteredViewUuids);
-        addComponentsBelowView(builder, session, authors, projects, projectUuids, moduleUuids, directories, fileUuids);
-      } else if ("DEV".equals(uniqueQualifier)) {
-        // XXX No constant for developer !!!
-        Collection<String> actualAuthors = authors == null ? dbClient.authorDao().selectScmAccountsByDeveloperUuids(allComponentUuids) : authors;
-        addComponentsBelowView(builder, session, actualAuthors, projects, projectUuids, moduleUuids, directories, fileUuids);
-      } else if (Qualifiers.PROJECT.equals(uniqueQualifier)) {
-        builder.projectUuids(allComponentUuids);
-        addComponentsBelowModule(builder, moduleUuids, directories, fileUuids);
-      } else if (Qualifiers.MODULE.equals(uniqueQualifier)) {
-        builder.moduleRootUuids(allComponentUuids);
-        addComponentsBelowModule(builder, moduleUuids, directories, fileUuids);
-      } else if (Qualifiers.DIRECTORY.equals(uniqueQualifier)) {
-        Collection<String> directoryModuleUuids = Sets.newHashSet();
-        Collection<String> directoryPaths = Sets.newHashSet();
-        for (ComponentDto directory : componentService.getByUuids(session, allComponentUuids)) {
-          directoryModuleUuids.add(directory.moduleUuid());
-          directoryPaths.add(directory.path());
-        }
-        builder.moduleUuids(directoryModuleUuids);
-        builder.directories(directoryPaths);
-        addComponentsBelowDirectory(builder, fileUuids);
-      } else if (Qualifiers.FILE.equals(uniqueQualifier) || Qualifiers.UNIT_TEST_FILE.equals(uniqueQualifier)) {
-        builder.fileUuids(allComponentUuids);
-      } else {
-        throw new IllegalArgumentException("Unable to set search root context for components " + Joiner.on(',').join(allComponentUuids));
-      }
-    }
-  }
-
-  private boolean mergeComponentParameters(DbSession session, Boolean onComponentOnly,
-    Collection<String> components,
-    Collection<String> componentUuids,
-    Collection<String> componentKeys,
-    Collection<String> componentRootUuids,
-    Collection<String> componentRoots,
     Set<String> allComponentUuids) {
     boolean effectiveOnComponentOnly = false;
 
+    failOnIncompatibleParameters(components, componentUuids, componentKeys, componentRootUuids, componentRoots);
```

Comment:
This method should be removed : it adds 10 lines, but it can be replaced by only 3 lines...